src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -728,31 +728,21 @@ public void SetLocation(Geometry geometry)
         {
             TransactionManager.Instance.EnsureInTransaction(Application.Document.Current.InternalDocument);
 
-            if (this.InternalElement.Location.GetType() == typeof(Autodesk.Revit.DB.LocationPoint))
+            if (this.InternalElement.Location.GetType() == typeof(Autodesk.Revit.DB.LocationPoint) && typeof(Autodesk.DesignScript.Geometry.Point).IsAssignableFrom(geometry.GetType()))
```

Comment:
You can include casting for curves also at the beginning. Also instead of checking IsAssignable, or equality for types, you can directly use `if(geometry is Point){..}`